symengine/functions.cpp

```diff
@@ -1216,33 +1216,33 @@ RCP<const Basic> ACsc::create(const RCP<const Basic> &arg) const
 Log::Log(const RCP<const Basic> &arg) : OneArgFunction(arg)
 {
     SYMENGINE_ASSIGN_TYPEID()
-    SYMENGINE_ASSERT(is_canonical(*arg))
+    SYMENGINE_ASSERT(is_canonical(arg))
 }
 
-bool Log::is_canonical(const Basic &arg) const
+bool Log::is_canonical(const RCP<const Basic> &arg) const
```

Comment:
All other function's `is_canonical()` takes `const RCP<const Basic> &` as argument type.To maintain consistency , i changed this. Is there any particular reason why only `Log`  has `const Basic &` as argument type ?